src/qubx/impl/ccxt_connector.py

```diff
@@ -155,34 +167,34 @@ async def _listen_to_execution_reports(self, channel: CtrlChannel, symbol: str):
                 _msg = f"\nexecs_{symbol} = [\n"
                 for report in exec:
                     _msg += "\t" + str(report) + ",\n"
-                    order, deals = self._process_execution_report(symbol, report)
+                    order, deals = self.trading_service.process_execution_report(symbol, report)
                     # - send update to client
-                    channel.queue.put((symbol, "order", order))
+                    channel.send((symbol, "order", order))
                     if deals:
-                        channel.queue.put((symbol, "deals", deals))
+                        channel.send((symbol, "deals", deals))
                 logger.debug(_msg + "]\n")
 
             except NetworkError as e:
-                logger.error(f"(CCXTConnector) NetworkError in _listen_to_execution_reports : {e}")
+                logger.error(f"(CCXTDataConnector) NetworkError in _listen_to_execution_reports : {e}")
                 await asyncio.sleep(1)
                 continue
 
             except ExchangeClosedByUser:
                 # - we closed connection so just stop it
-                logger.info(f"(CCXTConnector) {symbol} execution reports listening has been stopped")
+                logger.info(f"(CCXTDataConnector) {symbol} execution reports listening has been stopped")
                 break
 
             except Exception as err:
-                logger.error(f"(CCXTConnector) exception in _listen_to_execution_reports : {err}")
-                logger.error(stackprinter.format(err))
+                logger.error(f"(CCXTDataConnector) exception in _listen_to_execution_reports : {err}")
+                logger.exception(err)
 
     async def _listen_to_ohlcv(self, channel: CtrlChannel, symbol: str, timeframe: str, nbarsback: int):
         # - check if we need to load initial 'snapshot'
         if nbarsback >= 1:
             start = self._time_msec_nbars_back(timeframe, nbarsback)
             ohlcv = await self._exchange.fetch_ohlcv(symbol, timeframe, since=start, limit=nbarsback + 1)  # type: ignore
```

Comment:
I'm trying to run your example live paper trading strategy and it doesn't work with nback>0. For some reason it can't fetch candles, do you know the reason why?

Reply:
this is resolved, it was because of the self.trading_service.time from the other comment